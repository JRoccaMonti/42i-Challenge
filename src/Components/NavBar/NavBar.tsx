import style from "./navbar.module.css";
export const NavBar = () => {
    return (
      <div className={style.root}>
        <h3>42i Chalenge</h3>
        <a href="coverage\lcov-report\index.html">Jest Coverage</a>
      </div>
    );
};