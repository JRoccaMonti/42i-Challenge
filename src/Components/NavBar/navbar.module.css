.root{
    background-color: #6e0d0d;
    width: 100vw;
    position: fixed;
    opacity: 0.95;
    top: 0;
    left: 0; 
    display: flex;
    justify-content: space-between;
    align-items: center;
    padding: 0 25px;
    box-sizing: border-box
}

.jestLink {
    background-color: #9b2c2c;
    color: white;
    padding: 10px 20px;
    text-align: center;
    text-decoration: none;
    display: inline-block;
    border-radius: 5px;
    border: none;
    font-size: 16px;
    cursor: pointer;
    transition: background-color 0.3s ease;
  }
  
  .jestLink:hover {
    background-color: #b33a3a;
  }