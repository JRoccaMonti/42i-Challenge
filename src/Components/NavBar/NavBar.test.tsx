import { render, screen  } from '@testing-library/react';
import '@testing-library/jest-dom'
import {NavBar} from './NavBar';

describe('NavBar Component', () => {
  it('debería coincidir con la snapshot', () => {
    const { asFragment } = render(<NavBar />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('debería renderizar correctamente el título y el enlace', () => {
    render(<NavBar />);
    const titleElement = screen.getByText(/42i Chalenge/i);
    expect(titleElement).toBeInTheDocument();
    const linkElement = screen.getByText(/Jest Coverag/i);
    expect(linkElement).toBeInTheDocument();
    expect(linkElement).toHaveAttribute('href', 'coverage\\lcov-report\\index.html');
  });
});
