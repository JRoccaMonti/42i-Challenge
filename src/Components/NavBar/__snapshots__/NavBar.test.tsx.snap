// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar Component deber√≠a coincidir con la snapshot 1`] = `
<DocumentFragment>
  <div
    class="root"
  >
    <h3>
      42i Chalenge
    </h3>
    <a
      href="coverage\\lcov-report\\index.html"
    >
      Jest Coverage
    </a>
  </div>
</DocumentFragment>
`;
